test/e2e/upgradetest/resize_test.go

```diff
@@ -29,6 +29,9 @@ func TestResize(t *testing.T) {
 		if err != nil {
 			t.Fatal(err)
 		}
+		if err = e2eutil.WaitUntilDeploymentDeleted(testF.KubeCli, 3*time.Second, 30*time.Second, "etcd-operator", testF.KubeNS); err != nil {
```

Comment:
The pod name is not unique. Moreover, there should not be two existing operator pods at any give time because they manage the same namespace.

My experience with `kubectl delete deployment` is that it will return after deployment and its pods are deleted. We might want the same behavior. The question is if we change grace period to 1 for deployment, what would happen? Is Delete() call sync? Otherwise how is kubectl implemented?